controllers/kubeadmconfig_controller.go

```diff
@@ -162,14 +162,12 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 		}
 	}()
 
-	holdLock := false
-	defer func() {
-		if !holdLock {
-			r.KubeadmInitLock.Unlock(ctx, cluster)
-		}
-	}()
-
 	if !cluster.Status.ControlPlaneInitialized {
+		defer func() {
+			if rerr != nil {
```

Comment:
I'm interested in this change, why do we only unlock when rerr has an error?

Reply:
Since any error case in this block means the bootstrap data is not generated successfully, we restore the lock. Do you have any better suggestion?

Reply:
@chuckha I have moved this code behide the Lock code.